packages/template-typescript-minimal/app/pages/home.tsx

```diff
@@ -105,6 +109,18 @@ const Home = ({value}: Props) => {
         return () => clearInterval(interval)
     }, [counter, setCounter])
 
+    useServerContext(({res: pageRes}) => {
+        // Because the component is rendered twice on the server, this callback will also be called twice:
+        // 1. initially when the query is sent
+        // 2. and then when a response (or error) comes back
+
+        console.log('--- query', {isLoading: query.isLoading, error: query.error, data: query.data})
+
+        if (query.error) {
+            pageRes.status(404)
+        }
+    })
```

Comment:
Example of using both `useServerContext` and `useQuery`

Reply:
@bendvc that's expected, right? For the component to be rendered twice on the server.

Reply:
We aren't doing 2 renders per-say like we were going in the `useServerHook` implementation, but we are using the `react-ssr-prepass` library to get all the queries being run from ReactQuery. I'm 99% sure that is also the reason why the useServerContext callback function is getting called twice.

I'm guessing that the condition in your hook that checks for the request object in the context value prevents the callback from being called during that pre-pass, correct?

Reply:
I've just checked: yeah, both request and response exist during those passes.